apps/src/lib/levelbuilder/lesson-editor/VocabulariesEditor.jsx

```diff
@@ -0,0 +1,161 @@
+import PropTypes from 'prop-types';
+import React, {Component} from 'react';
+import {vocabularyShape} from '@cdo/apps/lib/levelbuilder/shapes';
+import color from '@cdo/apps/util/color';
+import SearchBox from './SearchBox';
+import Dialog from '@cdo/apps/templates/Dialog';
+import {connect} from 'react-redux';
+import {
+  addVocabulary,
+  editVocabulary,
+  removeVocabulary
+} from '@cdo/apps/lib/levelbuilder/lesson-editor/vocabulariesEditorRedux';
+
+const styles = {
+  oddRow: {
+    backgroundColor: color.lightest_gray
+  },
+  actionsColumn: {
+    display: 'flex',
+    justifyContent: 'space-evenly',
+    backgroundColor: 'white'
+  },
+
+  remove: {
+    fontSize: 14,
+    color: 'white',
+    background: color.dark_red,
+    cursor: 'pointer',
+    textAlign: 'center',
+    width: '98%'
+  }
+};
+
+class VocabulariesEditor extends Component {
+  static propTypes = {
+    courseVersionId: PropTypes.number,
+
+    // Provided by redux
+    vocabularies: PropTypes.arrayOf(vocabularyShape).isRequired,
+    addVocabulary: PropTypes.func.isRequired,
+    editVocabulary: PropTypes.func.isRequired,
+    removeVocabulary: PropTypes.func.isRequired
+  };
+
+  constructVocabularyOption = vocabulary => ({
+    value: vocabulary.key,
+    label: `${vocabulary.word} - ${vocabulary.definition}`,
+    vocabulary
+  });
+
+  constructor(props) {
+    super(props);
+    this.state = {confirmRemovalDialogOpen: false, vocabularyForRemoval: null};
+  }
+
+  constructSearchOptions = json => {
+    const vocabKeysAdded = this.props.vocabularies.map(vocab => vocab.key);
+    const vocabularies = json
+      .map(vocab => this.constructVocabularyOption(vocab))
+      .filter(vocab => vocabKeysAdded.indexOf(vocab.value) === -1);
+    return {options: vocabularies};
+  };
+
+  handleRemoveVocabularyDialogClose = () => {
+    this.setState({
+      confirmRemovalDialogOpen: false,
+      vocabularyForRemoval: null
+    });
+  };
+
+  handleRemoveVocabularyConfirm = () => {
+    this.props.removeVocabulary(this.state.vocabularyForRemoval.key);
+    this.handleRemoveVocabularyDialogClose();
+  };
+
+  render() {
+    return (
+      <div>
+        {this.state.confirmRemovalDialogOpen && (
+          <Dialog
+            body={`Are you sure you want to remove the vocabulary "${
+              this.state.vocabularyForRemoval.word
+            }" from this lesson?`}
+            cancelText="Cancel"
+            confirmText="Delete"
+            confirmType="danger"
+            isOpen={this.state.confirmRemovalDialogOpen}
+            handleClose={this.handleRemoveVocabularyDialogClose}
+            onCancel={this.handleRemoveVocabularyDialogClose}
+            onConfirm={this.handleRemoveVocabularyConfirm}
+          />
+        )}
+        <input
```

Comment:
Is this input needed with the new ajax save process (and if not do we need to remove it in resource editor too)?

Reply:
Yeah I was wondering that. I don't think it does